go/libraries/doltcore/doltdb/tag_meta.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package doltdb
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+const (
+	tagMetaNameKey      = "name"
+	tagMetaEmailKey     = "email"
+	tagMetaDescKey      = "desc"
+	tagMetaTimestampKey = "timestamp"
+	tagMetaVersionKey   = "metaversion"
+
+	tagMetaStName  = "metadata"
+	tagMetaVersion = "1.0"
+)
+
+var TagNowFunc = CommitNowFunc
+var TagLoc = CommitLoc
+
+// TagMeta contains all the metadata that is associated with a tag within a data repo.
+type TagMeta struct {
+	Name        string
+	Email       string
+	Timestamp   uint64
+	Description string
+}
+
+// NewTagMeta returns CommitMeta that can be used to create a tag.
+func NewTagMeta(name, email, desc string) *TagMeta {
```

Comment:
Seems a little weird to me that this doesn't take the `time.Time` as well. But maybe this is fine...? In dolt's architecture maybe I would expect the following:

```
   On Backdating Tags
       If you have imported some changes from another VCS and would like to add tags for major releases of your work, it is useful
       to be able to specify the date to embed inside of the tag object; such data in the tag object affects, for example, the
       ordering of tags in the gitweb interface.

       To set the date used in future tag objects, set the environment variable GIT_COMMITTER_DATE (see the later discussion of
       possible values; the most common form is "YYYY-MM-DD HH:MM").

       For example:

           $ GIT_COMMITTER_DATE="2006-10-02 10:31" git tag -s v1.0.1
```

would be implemented through a `dEnv` lookup or something?

Reply:
I added user timestamp to the meta struct. There's currently no way to pass it from the CLI, but the model is there